### PR TITLE
Add Frames support to sequence Combine

### LINKED ISSUE #10469: Add Frames: support to Combined sub-sequences
MechCommander sprites use one .shp file for each facing, so using `Combine` is a must.

However, probably to save memory or at least HDD space, they usually only cover half of the facing 'clock' and require considerable amounts of flipping and reverse-playing.
FlipX/Y are supported on `Combine`d sub-sequences, but `Frames` and `Reverse` are not. It would be extremely useful (and for some animations even necessary) to add at least `Frames` support to `Combine`d sub-sequences, to make move animations for turrets and barrels easier to implement.

Basically, I'd like our code to support something like this: 
https://gist.github.com/reaperrr/63900af4afe9a560e762

### PR DESCRIPTION
**Pull request by reaperrr**
Interim solution for #10469 until the alternative proposed in https://github.com/OpenRA/OpenRA/issues/10469#issuecomment-590105685 gets implemented.

### CODE REVIEW

File: OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs [hunk 1]
```diff
@@ -261,27 +261,33 @@ public DefaultSpriteSequence(ModData modData, TileSet tileSet, SpriteCache cache
...
 						{
 							MiniYaml subLengthYaml;
 							if (sd.TryGetValue("Length", out subLengthYaml) && subLengthYaml.Value == "*")
-								subLength = subFrameCount - subStart;
+								subLength = subFrames != null ? subFrames.Length : subFrameCount - subStart;
 							else
 								subLength = LoadField(sd, "Length", 1);
 
-							return Enumerable.Range(subStart, subLength);
+							var subUsedFrames = new List<int>();
```
Comment (pchote) [inline OpenRA.Mods.Common/Graphics/DefaultSpriteSequence.cs]: Can we change this to `return subFrames != null ? subFrames.Skip(subStart).Take(subLength) : Enumerable.Range(subStart, subLength);`?

This is more concise and correctly accounts for sub-sequences that define custom Start/Length + Frames.

**Review by pchote — COMMENTED** [review]
LGTM. Just one simplification:

**Comment by reaperrr** [comment]
Updated.

**Review by pchote — DISMISSED** [review]

**Comment by reaperrr** [comment]
Removed testcase, since both myself and @pchote confirmed that it works.

If the next reviewer still needs one, just add
`Frames: 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0`
to any of the `flame-` or `chem-` definitions in the TD infantry sequences and let the infantry fire in that direction, you'll see a reversed flame animation.

For regression testing, the only other cases in our mods that use `Combine` are the TD Guard Tower make anim and TS large visceroid attack, both to skip certain frames.

**Comment by pchote** [comment]
This is a simple and low risk PR that naturally fits with the other Frames fixes, so adding to the milestone.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/89622cbfe1d747dee2311d984b8ee34967307ffd...5bd691833d37b25573ff85c77cc216ef563cd461)

**Pull request merged**